### PR TITLE
[Quant][Inductor] Enable qlinear weight prepack inside inductor constant folding

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #106935
* #106934
* __->__ #106782
* #106781
* #105818

**Summary**
To realize weight prepack for quantized linear, we replace the following pattern
```
int8 activation
      |
dequant_per_tensor
      |
mm/addmm <- t <- dequant_per_channel <- int8_weight
```
with
```
int8 activation
  |
onednn.qlinear_pointwise <- onednn.qlinear_prepack <- int8_weight
```
And we register weight prepack path inside inductor constant folding. Constant folding evaluates the prepack op and replace it with prepacked weight (a constant parameter)

**Test plan**
python test/inductor/test_mkldnn_pattern_matcher.py -k test_qlinear_unary

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/fx_passes/quantization.py [hunk 1]
```diff
@@ -588,6 +588,149 @@ def _generate_qconv_weight_prepack_patterns():
...
+                qx,
+                x_scale,
+                x_zp,
+                prepack_weight_node,
+                w_scale,
+                w_zp,
+                bias,
+                1.0,  # output_scale
+                0,  # output_zero_point
+                True,  # fp32_output
```
Comment (jgong5) [inline torch/_inductor/fx_passes/quantization.py:688]: we always output fp32 for qlinear?

Comment (Xia-Weiwen) [reply torch/_inductor/fx_passes/quantization.py:688]: Yes. This is for weight prepack, not the final graph. The original pattern outputs fp32, so we need to keep this semantics.

**Review by jgong5 — APPROVED** [review]

**Review by leslie-fang-intel — APPROVED** [review]

**Review by eellison — APPROVED** [review]

**Pull request closed without merging**